src/components/plan/PlanMergeReview.test.ts

```diff
@@ -0,0 +1,293 @@
+      initialNonConflictingActivities,
+      initialPlan: { ...mockInitialPlan },
+    });
+
+    expect(component).toBeTruthy();
+  });
+
+  it('PlanMergeReview component should not throw with conflicting activities when source is modify and target is delete', () => {
+    const initialConflictingActivities: PlanMergeConflictingActivity[] = [
+      {
+        activity_id: 1,
+        change_type_source: 'modify',
+        change_type_target: 'delete',
+        merge_base: {
+          anchor_id: null,
+          anchored_to_start: true,
+          arguments: {},
+          created_at: '2023-02-16T17:48:26',
+          id: 1,
+          last_modified_arguments_at: '2023-02-16T17:48:26',
+          last_modified_at: '2023-02-16T17:48:26',
+          metadata: {},
+          name: 'A_Activity',
+          plan_id: 1,
+          snapshot_id: 1,
+          source_scheduling_goal_id: null,
+          start_offset: '23:06:17.622',
+          tags: [],
+          type: 'A_Activity',
+        },
+        merge_request_id: 1,
+        resolution: 'none',
+        source: {
+          anchor_id: null,
+          anchored_to_start: true,
+          arguments: {},
+          created_at: '2023-02-16T17:48:26',
+          id: 2,
+          last_modified_arguments_at: '2023-02-16T17:48:26',
+          last_modified_at: '2023-02-16T17:48:47',
+          metadata: {},
+          name: 'A_Activity',
+          snapshot_id: 2,
+          source_scheduling_goal_id: null,
+          start_offset: '36:25:10.489',
+          tags: [],
+          type: 'A_Activity',
+        },
+        target: null,
+      },
+    ];
+    const initialNonConflictingActivities: PlanMergeNonConflictingActivity[] = [];
+
+    const { component } = render(PlanMergeReview, {
+      initialConflictingActivities,
+      initialMergeRequest: { ...mockMergeRequest },
+      initialNonConflictingActivities,
+      initialPlan: { ...mockInitialPlan },
+    });
+
+    expect(component).toBeTruthy();
+  });
+
+  it('PlanMergeReview component should not throw with non-conflicting activities', () => {
+    const initialConflictingActivities: PlanMergeConflictingActivity[] = [];
+    const initialNonConflictingActivities: PlanMergeNonConflictingActivity[] = [
+      {
+        activity_id: 6,
+        change_type: 'add',
+        merge_request_id: 1,
+        source: {
+          anchor_id: null,
+          anchored_to_start: true,
+          arguments: {},
+          created_at: '2023-02-16T20:41:07',
+          id: 6,
+          last_modified_arguments_at: '2023-02-16T20:41:07',
+          last_modified_at: '2023-02-16T20:41:07',
+          metadata: {},
+          name: 'B_Activity',
+          snapshot_id: 6,
+          source_scheduling_goal_id: null,
+          start_offset: '46:33:39.909',
+          tags: [],
+          type: 'B_Activity',
+        },
+        target: null,
+      },
+      {
+        activity_id: 7,
+        change_type: 'none',
```

Comment:
`change_type` would be "delete" here

Reply:
Can you go over the set up for this test? I just made a delete example and got from GraphQL:

```json
"get_non_conflicting_activities": [
      {
        "activity_id": 2,
        "change_type": "delete",
        "source": null,
        "target": {
          "id": 2,
          "plan_id": 4,
          "name": "ChangeProducer",
          "tags": [],
          "source_scheduling_goal_id": null,
          "created_at": "2023-02-16T21:26:42.515938+00:00",
          "last_modified_at": "2023-02-16T21:26:42.515938+00:00",
          "start_offset": "04:03:29.63",
          "type": "ChangeProducer",
          "arguments": {},
          "last_modified_arguments_at": "2023-02-16T21:26:42.515938+00:00",
          "metadata": {},
          "anchor_id": null,
          "anchored_to_start": true
        }
      }
    ]
```
(Also, `merge_request_id` was not a field I could return)

Reply:
Here is the test setup:
1. Create a plan with two activities
2. Branch the plan
3. Add a third activity to the branch
4. Merge the brach into the parent

Reply:
Yeah I think the type is wrong for `PlanMergeNonConflictingActivity`, it includes `merge_request_id` when it should not. Will remove.

Reply:
Repeating those steps, `source` was not `null` for either of the `change_type=none`s (Which is expected). Wonder why it's null here.

Reply:
> Yeah I think the type is wrong for `PlanMergeNonConflictingActivity`, it includes `merge_request_id` when it should not. Will remove.

Then `PlanMergeConflictingActivity` probably shouldn't have a `merge_request_id` either.